core/bootstrap/src/main/java/org/eclipse/dataspaceconnector/system/readiness/ReadinessServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package org.eclipse.dataspaceconnector.system.readiness;
+
+import org.eclipse.dataspaceconnector.spi.system.readiness.LivenessProvider;
+import org.eclipse.dataspaceconnector.spi.system.readiness.ReadinessProvider;
+import org.eclipse.dataspaceconnector.spi.system.readiness.ReadinessService;
+import org.eclipse.dataspaceconnector.spi.system.readiness.StartupStatusProvider;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class ReadinessServiceImpl implements ReadinessService {
+    private final List<LivenessProvider> livenessProviders;
+    private final List<ReadinessProvider> readinessProviders;
+    private final List<StartupStatusProvider> startupStatusProviders;
+
+    public ReadinessServiceImpl() {
+        readinessProviders = new CopyOnWriteArrayList<>();
+        livenessProviders = new CopyOnWriteArrayList<>();
+        startupStatusProviders = new CopyOnWriteArrayList<>();
+    }
+
+    @Override
+    public void addLivenessProvider(LivenessProvider provider) {
+        livenessProviders.add(provider);
+    }
+
+    @Override
+    public void addReadinessProvider(ReadinessProvider provider) {
+        readinessProviders.add(provider);
+    }
+
+    @Override
+    public void addStartupStatusProvider(StartupStatusProvider provider) {
+        startupStatusProviders.add(provider);
+    }
+
+    @Override
+    public boolean isLive() {
+        return livenessProviders.stream().allMatch(LivenessProvider::get);
```

Comment:
Logging would be useful for failing providers. In consequence consider using a loop without short-circuit to ensure all failing providers are logged, not just the first one.